src/wallet/wallet.cpp

```diff
@@ -3670,6 +3715,27 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
 
         walletInstance->SetBestChain(chainActive.GetLocator());
     }
+    else if (IsArgSet("-disablehot")) {
+        bool disableHot = GetBoolArg("-disablehot", DEFAULT_DISABLE_HOT_WALLET);
+        LOCK(walletInstance->cs_wallet);
+        if (walletInstance->vchDefaultKey.IsValid() && disableHot) {
```

Comment:
Indeed. Might be out of this PR matter, in the case of hdwatchonly, `walletInstance->vchDefaultKey.IsValid()` will be true. So this would mean `disabledHot` is incompatible with hdwatchonly mode ?